### ISSUE TITLE
Could not determinte it's EKS cluster MissingRegion: could not find region configuration

### ISSUE THREAD

**Issue by apryiomka** [kubeflow/kubeflow#4854]
/kind bug

**What steps did you take and what happened:**
Trying to install Kubeflow in AWS EKS cluster using kfctl apply -f kfctl_aws.yaml

**What did you expect to happen:**
Kubeflow is successfully installed

**Anything else you would like to add:**
Running this inside docker container

**Environment:**

- Kubeflow version: 1.0.1:
- kfctl version: (use `kfctl version`): 1.0.1

I am getting an error:

> kfctl apply --verbose --file kfctl_aws.1.0.1.yaml
 time="2020-03-11T22:30:20Z" level=info msg="Running `eksctl version` ..." filename="utils/awsutil.go:71"
 time="2020-03-11T22:30:20Z" level=info msg="eksctl version: 0.14.0" filename="utils/awsutil.go:88"
 time="2020-03-11T22:30:20Z" level=info msg="\n****************************************************************\nNotice anonymous usage reporting enabled using spartakus\nTo disable it\nIf you have already deployed it run the following commands:\n  cd $(pwd)\n  kubectl -n ${K8S_NAMESPACE} delete deploy -l app=spartakus\n\nFor more info: https://www.kubeflow.org/docs/other-guides/usage-reporting/\n****************************************************************\n\n" filename="coordinator/coordinator.go:120"
 time="2020-03-11T22:30:20Z" level=info msg=".cache/manifests exists; not resyncing " filename="kfconfig/types.go:468"
 time="2020-03-11T22:30:20Z" level=info msg="Folder aws_config exists, skip aws.Generate" filename="aws/aws.go:438"
 time="2020-03-11T22:30:21Z" level=info msg="folder kustomize exists, skip kustomize.Generate" filename="kustomize/kustomize.go:372"
 time="2020-03-11T22:30:21Z" level=info msg=".cache/manifests exists; not resyncing " filename="kfconfig/types.go:468"
 time="2020-03-11T22:30:21Z" level=info msg="Caller ARN Info: {\n  Account: \"277749501672\",\n  Arn: \"arn:aws:sts::111111111:assumed-role/service-deploy/service-deploy\",\n  UserId: \"AAAAAAAA-deploy\"\n}" filename="utils/awsutil.go:41"
 time="2020-03-11T22:30:21Z" level=info msg="ManagedCluster set defaulting to false" filename="aws/aws.go:651"
 time="2020-03-11T22:30:21Z" level=info msg="EnablePodIamPolicy set defaulting to false" filename="aws/aws.go:656"
 time="2020-03-11T22:30:21Z" level=info msg="EnableNodeGroupLog set defaulting to false" filename="aws/aws.go:661"
 time="2020-03-11T22:30:21Z" level=info msg="You already have cluster setup. Skip creating new eks cluster. " filename="aws/eks.go:95"
 Error: failed to apply:  (kubeflow.error): Code 500 with message: coordinator Apply failed for aws:  (kubeflow.error): Code 400 with message: Could not determinte it's EKS cluster MissingRegion: could not find region configuration

This is my cluster config:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: build
  region: us-west-2
  version: "1.14"
nodeGroups:
- desiredCapacity: 2
  instanceType: m5.xlarge
  maxSize: 3
  minSize: 0
  name: cpu-nodegroup
  volumeSize: 30
```

Also the IAM user the deployment runs under has the region configured:
`export AWS_DEFAULT_REGION=us-west-2`

**Comment by Julianxo** [comment]
Have you set a region in your kfctl_aws.yaml? (Last line in cutout)

`  plugins:
  - kind: KfAwsPlugin
    metadata:
      name: aws
    spec:
      auth:
        basicAuth:
          password:
            name: password
          username: admin
      region: us-west-2`

Like you find it here in the default config:
https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_aws.v1.0.1.yaml

**Comment by kslin** [comment]
I got this error as well using kubeflow 1.0.1, which I didn't get using kubeflow 1.0 (I double checked that I set a region in my kfctl_aws.yaml file). I'm using eksctl 0.14.0 and kubernetes 1.14.7.

**Comment by apryiomka** [comment]
@Julianxo the region is set correclty:

```
plugins:
  - kind: KfAwsPlugin
    metadata:
      creationTimestamp: null
      name: aws
    spec:
      auth:
        basicAuth:
          password:
            name: password
          username: admin
      region: us-west-2
      roles:
      - blah-blah-blah
```

**Comment by apryiomka** [comment]
@kslin Unfortunately, i cannot use 1.0.0 as it doesn't have extarnal-mysql overlays for metastore and pipelines.

**Comment by apryiomka** [comment]
I am bit confused by the `name` field in the cluster config though:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: manifest
  region: us-west-2
  version: "1.14"
nodeGroups:
- desiredCapacity: 2
  instanceType: m5.xlarge
  maxSize: 3
  minSize: 0
  name: cpu-nodegroup
  volumeSize: 30
```

Isn't the name supposed to be the cluster name? I am runing `kfctl` as so:
```
sed -i'.bak' -e "s/kubeflow-iam-role/$worker_node_role/" ${config_file}

export KF_NAME=${aws_cluster_name}
kfctl build --verbose --file ${config_file}
```

**Comment by Julianxo** [comment]
@kslin 
> I got this error as well using kubeflow 1.0.1, which I didn't get using kubeflow 1.0 (I double checked that I set a region in my kfctl_aws.yaml file). I'm using eksctl 0.14.0 and kubernetes 1.14.7.
Then I guess it is a bug in version 1.0.1 and @apryiomka is doing nothing wrong. But I am not a kubeflow developer, maybe one is here? 

@apryiomka 
Why are you running kfctl build? The new documentation of kubeflow v1.0 or newer does not suppose to use this command as you can see here https://www.kubeflow.org/docs/aws/deploy/install-kubeflow/.

Just use **wget** to fetch your config, do your changes and run kfctl apply -f YOUR_CONFIG_FILE afterwards. By using version 1.0 with kfctl build I had some strange problems too.

**Comment by apryiomka** [comment]
@Julianxo it is not the build, it is the _official_ release / tag: https://github.com/kubeflow/kfctl/releases/download/v1.0.1/kfctl_v1.0.1-0-gf3edb9b_darwin.tar.gz

it is coming from there: https://github.com/kubeflow/kfctl/releases/tag/v1.0.1

Somehow i assumed to use the 1.0.1 manifest i need 1.0.1 kctl, is it not the case?

**Comment by fsaina** [comment]
Set the region of your aws cli and it will work:

`aws configure set default.region us-west-2`

But this is a bug because kfctl completely ignores the region set in the .yaml file and just uses the aws cli default.

**Comment by sssyyw** [comment]
> Set the region of your aws cli and it will work:
> 
> `aws configure set default.region us-west-2`
> 
> But this is a bug because kfctl completely ignores the region set in the .yaml file and just uses the aws cli default.

@fsaina, I ran into the same issue. Setting the region using aws cli does not work for me...

**Comment by Jeffwan** [comment]
```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: manifest
  region: us-west-2
  version: "1.14"
nodeGroups:
- desiredCapacity: 2
  instanceType: m5.xlarge
  maxSize: 3
  minSize: 0
  name: cpu-nodegroup
  volumeSize: 30

```

Hi all, please don't be confused about this. This was used to created a kubeflow with a new EKS cluster. Now, the recommend path is to deploy kubeflow on existing EKS cluster which means this file is not helpful.

```
Error: failed to apply: (kubeflow.error): Code 500 with message: coordinator Apply failed for aws: (kubeflow.error): Code 400 with message: Could not determinte it's EKS cluster MissingRegion: could not find region configuration
```
 This indicates that is failed here. 

https://github.com/kubeflow/kfctl/blob/1fe560ef9bdb3e8aac135f0e7ff45e7d097bdf17/pkg/kfapp/aws/aws.go#L710-L716

it uses default resolver to make a call to EKS 
https://github.com/kubeflow/kfctl/blob/1fe560ef9bdb3e8aac135f0e7ff45e7d097bdf17/pkg/kfapp/aws/eks.go#L49-L65

Please make sure you have AWS_REGION environment set in your environment to mitigate this issue temporarily . We will file a PR improvement to use plugin's region user filled to create a SDK session, or resolve profile provider properly. 

https://docs.aws.amazon.com/es_es/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-the-region

**Comment by Jeffwan** [comment]
@apryiomka @fsaina @sssyyw @kslin @Julianxo 

The problem we have more issues in 1.0.1 is because we use isEKscluster to check if we apply some EKS specific logic like IAM for service account, etc.

**Comment by fsaina** [comment]
@sssyyw just tried it, you are right it's not the aws cli tool config, it's the AWS_REGION env variable. After that it works on 1.0.1, for me at least.

`export AWS_REGION=eu-central-1`

**Comment by apryiomka** [comment]
@Jeffwan setting default aws region didn't work for me, please re-read my very first comment

> Also the IAM user the deployment runs under has the region configured:
> `export AWS_DEFAULT_REGION=us-west-2`

[Unless there is an undocumented feature](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html): 

> AWS_DEFAULT_REGION
> Specifies the AWS Region to send the request to.
> 
> If defined, this environment variable overrides the value for the profile setting region. You can override this environment variable by using the --region command line parameter.

**Comment by PatrickXYS** [comment]
Hi all, a quick workaround for this issue is:
```
export AWS_SDK_LOAD_CONFIG=1 
```
This can make session read configuration information from config file (`~/.aws/config`).

**Comment by PatrickXYS** [comment]
@apryiomka Since according to the [AWS SDK Go doc](https://docs.aws.amazon.com/es_es/sdk-for-go/v1/developer-guide/configuring-sdk.html), we should set the region with 
```
export AWS_REGION=us-west-2
```
For reference: 
![image](https://user-images.githubusercontent.com/23116624/76907782-562c8d00-6864-11ea-9844-a4bc74390142.png)

But this is just for workaround, I'll fix it in the PR and you can see it in the next patch revision of `kfctl`.

**Comment by Jeffwan** [comment]
@PatrickXYS ,Could you help cherry-pick the change to v1.0-branch?

**Comment by PatrickXYS** [comment]
@Jeffwan  Sure, I'm planning to do it currently.

**Comment by Jeffwan** [comment]
Let's wait for 1.0.2 or 1.1.0 release.

**Comment by aimldl** [comment]
> Hi all, a quick workaround for this issue is:
> 
> ```
> export AWS_SDK_LOAD_CONFIG=1 
> ```
> 
> This can make session read configuration information from config file (`~/.aws/config`).
Thanks to PatrickXYS, the following command fixed the error and Kubeflow installed successfully.

$ aws configure set default.region us-west-2
$ export AWS_SDK_LOAD_CONFIG=1 

Thanks again!

**Comment by Jeffwan** [comment]
This has been included in the v1.0-branch. It will be released this week in v1.0.2. I will close this issue. Feel free to reopen if problem still exist.

**Comment by majorinche** [comment]
> 
> 
> Hi all, a quick workaround for this issue is:
> 
> ```
> export AWS_SDK_LOAD_CONFIG=1 
> ```
> 
> This can make session read configuration information from config file (`~/.aws/config`).

this works for me, thanks!
